### PR TITLE
Add disable port discovery feature

### LINKED ISSUE #2794: Disable discovery on a specific port on a switch
As an admin, I can disable discovery on  a specific port on a switch

### PR DESCRIPTION
**Pull request by rozdy**
Closes #2794

### CODE REVIEW

**Comment by rozdy** [comment]
@surabujin 
> > ```java
> > @NoArgsConstructor
> > @JsonNaming(SnakeCaseStrategy.class)
> > public class PortPropertiesPayload extends InfoData {
> > ```
> 
> It will be better to put here `PortProperties` object from `kilda-model` package.
> 

PortProperties has reference to Switch object. Is it ok to serialize/deserialize whole Switch object when we need only switch id?

> > ```java
> >     @Mapping(source = "switchObj.switchId", target = "switchId")
> >     @Mapping(source = "port", target = "portNumber")
> >     public abstract PortPropertiesPayload map(PortProperties portInfoData);
> > ```
> 
> `portInfoData`?

Fixed.

> > ```java
> >         PortProperties portProperties = switchOperationsService.getPortProperties(request.getSwitchId(),
> >                 request.getPort());
> >         if (portProperties == null) {
> > ```
> 
> Why "response" logic live outside "service layer"?
> > ```java
> >             String message = String.format("Port properties not found for switch '%s' and port '%d'",
> >                     request.getSwitchId(), request.getPort());
> >             log.error(message);
> > ```
> 
> ??? O_o
> 
> This is the responsibility of the exception handler.

It was made for similarity reason: a lot of methods around has this logic on this layer. Should we move it somewhere?

> > ```java
> >             Optional<PortProperties> foundPortProperties =
> >                     portPropertiesRepository.getBySwitchIdAndPort(switchId, port);
> >             if (foundPortProperties.isPresent()) {
> > ```
> 
> Why for so ... complex. Revert this condition and create new instance inside... later, outside condition apply required update and save new/existing object into DB.

Update logic was moved to network topology and changed.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SwitchOperationsService.java [hunk 1]
```diff
@@ -147,6 +151,9 @@ public boolean deleteSwitch(SwitchId switchId, boolean force) throws SwitchNotFo
         Switch sw = switchRepository.findById(switchId)
                 .orElseThrow(() -> new SwitchNotFoundException(switchId));
 
+        if (sw.getPortProperties() != null) {
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SwitchOperationsService.java]: why is it not a single transaction for switch port properties and switch properties?

File: services/wfm/src/main/java/org/openkilda/wfm/topology/network/controller/port/PortFsm.java [hunk 2]
```diff
@@ -131,6 +150,10 @@ public void finish(PortFsmState from, PortFsmState to, PortFsmEvent event, PortF
                     .callMethod("proxyDiscovery");
             builder.internalTransition().within(PortFsmState.UP).on(PortFsmEvent.FAIL)
                     .callMethod("proxyFail");
+            builder.internalTransition().within(PortFsmState.UP).on(PortFsmEvent.ENABLE_DISCOVERY)
```
Comment (sergii-iakovenko) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/network/controller/port/PortFsm.java]: Does it mean that we can't enable/disable discovery for a port in down state?

**Comment by rozdy** [comment]
> _[services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/SwitchOperationsBolt.java, line 213 at r2](https://reviewable.io/reviews/telstra/open-kilda/2814#-LqBg5A71j_JxwcUw8Vp:-LqBg5A71j_JxwcUw8Vq:bxui6hl) ([raw file](https://github.com/telstra/open-kilda/blob/b36ec016c76a3022179ee70368448f2f336f4950/services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/SwitchOperationsBolt.java#L213)):_
> 
> > ```java
> >         } catch (PersistenceException e) {
> >             String message = String.format("Port properties not found: %s", e.getMessage());
> >             log.error(message);
> > ```
> 
> Remove this logger call. The exception handler can write something in the log if it want to.
> 

This exception is handled in PersistenceOperationsBolt and message is not logged. I guess we should refactor this separately.

@sergii-iakovenko @surabujin @timofei-durakov 
All other comments are fixed.

**Comment by rozdy** [comment]
> _[services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/SwitchOperationsBolt.java, line 213 at r1](https://reviewable.io/reviews/telstra/open-kilda/2814#-LpYUqH_6x2ECjgXwp5V:-LqLUwih-_UziSgrnOUi:b-fl6zpg) ([raw file](https://github.com/telstra/open-kilda/blob/acf47ca566f1548a09dd1ad331e22ac7991bcca0/services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/SwitchOperationsBolt.java#L213)):_
> Previously, surabujin (Dmitriy Bogun) wrote…
> 
> Bolt is a transport layer. Move this logic(handling the exception and producing some response on in) into service.
> 
> _[services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/SwitchOperationsBolt.java, line 213 at r2](https://reviewable.io/reviews/telstra/open-kilda/2814#-LqBg5A71j_JxwcUw8Vp:-LqLMCagD0b_y07MZk5t:bhs754y) ([raw file](https://github.com/telstra/open-kilda/blob/b36ec016c76a3022179ee70368448f2f336f4950/services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/SwitchOperationsBolt.java#L213)):_
> Previously, rozdy (Ievgen Rozdymakha) wrote…
> 
> Ok. Create issue/pr for this.
> 

Issue #2847 created

**Pull request merged**